<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainForm</class>
 <widget class="QMainWindow" name="MainForm">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>993</width>
    <height>655</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtTerrainGenerator</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="viewTabWidget">
    <property name="geometry">
     <rect>
      <x>5</x>
      <y>11</y>
      <width>800</width>
      <height>620</height>
     </rect>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Triangular</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="heightmapViewTab">
     <attribute name="title">
      <string>Heightmap View</string>
     </attribute>
     <widget class="QLabel" name="heightMapLabel">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>-1</y>
        <width>800</width>
        <height>600</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="terrainViewTab">
     <attribute name="title">
      <string>Terrain View</string>
     </attribute>
     <widget class="TerrainVisualizationWidget" name="terrainViewWidget" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>-1</y>
        <width>800</width>
        <height>600</height>
       </rect>
      </property>
      <property name="mouseTracking">
       <bool>false</bool>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QTabWidget" name="attributesTabWidget">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>11</y>
      <width>181</width>
      <height>620</height>
     </rect>
    </property>
    <property name="whatsThis">
     <string/>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="tabPosition">
     <enum>QTabWidget::North</enum>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Triangular</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="elideMode">
     <enum>Qt::ElideRight</enum>
    </property>
    <property name="usesScrollButtons">
     <bool>false</bool>
    </property>
    <property name="documentMode">
     <bool>false</bool>
    </property>
    <property name="tabsClosable">
     <bool>false</bool>
    </property>
    <property name="movable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="heightMapAttributesTab">
     <attribute name="title">
      <string>Heightmap Attributes</string>
     </attribute>
     <widget class="QFrame" name="frame_2">
      <property name="geometry">
       <rect>
        <x>8</x>
        <y>34</y>
        <width>159</width>
        <height>254</height>
       </rect>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>145</width>
         <height>232</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <property name="verticalSpacing">
         <number>10</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>12</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>A</string>
          </property>
          <property name="textFormat">
           <enum>Qt::AutoText</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="aSpinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
          <property name="value">
           <double>12.345670000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Si</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="sStartSpinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
          <property name="value">
           <double>0.567890000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_3">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>λ</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="lambdaSpinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_4">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>μ</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDoubleSpinBox" name="muSpinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_5">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>σ</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QDoubleSpinBox" name="sigmaSpinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_6">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>ε</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QDoubleSpinBox" name="epsilonSpinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_7">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>η</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QDoubleSpinBox" name="etaSpinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="label_8">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>γ</string>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QDoubleSpinBox" name="gammaSpinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QComboBox" name="randomGeneratorComboBox">
      <property name="geometry">
       <rect>
        <x>8</x>
        <y>8</y>
        <width>159</width>
        <height>20</height>
       </rect>
      </property>
      <property name="currentIndex">
       <number>-1</number>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QComboBox::AdjustToMinimumContentsLengthWithIcon</enum>
      </property>
      <property name="frame">
       <bool>true</bool>
      </property>
      <item>
       <property name="text">
        <string>Равномерные случайные числа</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Нормированные нормальные случайные числа</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Экспоненциальные случайные числа</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Логарифмически нормальные случайные числа</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Джонсона Sв</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Джонсона Su</string>
       </property>
      </item>
     </widget>
     <widget class="QFrame" name="frame">
      <property name="geometry">
       <rect>
        <x>8</x>
        <y>294</y>
        <width>159</width>
        <height>109</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>33</x>
         <y>6</y>
         <width>88</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Noise zoom factor</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="noiseZoomFactorSpinBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>121</width>
         <height>19</height>
        </rect>
       </property>
       <property name="minimum">
        <number>2</number>
       </property>
       <property name="maximum">
        <number>512</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>32</x>
         <y>55</y>
         <width>91</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Size of heightmap</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="SizeOfHeightmapSpinBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>121</width>
         <height>19</height>
        </rect>
       </property>
       <property name="minimum">
        <number>2</number>
       </property>
       <property name="maximum">
        <number>512</number>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="generateHeightmapButton">
      <property name="geometry">
       <rect>
        <x>8</x>
        <y>550</y>
        <width>159</width>
        <height>40</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate Heightmap</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="terrainAttributesTab">
     <attribute name="title">
      <string>Terrain Attributes</string>
     </attribute>
     <widget class="QFrame" name="frame_3">
      <property name="geometry">
       <rect>
        <x>7</x>
        <y>8</y>
        <width>159</width>
        <height>59</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>53</x>
         <y>6</y>
         <width>88</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Scale factor</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="scaleFactorSpinBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>121</width>
         <height>19</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
      </widget>
     </widget>
     <widget class="QFrame" name="frame_4">
      <property name="geometry">
       <rect>
        <x>7</x>
        <y>77</y>
        <width>159</width>
        <height>89</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QPushButton" name="travelButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>136</width>
         <height>40</height>
        </rect>
       </property>
       <property name="text">
        <string>Travel</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="fullscreenCheckBox">
       <property name="geometry">
        <rect>
         <x>45</x>
         <y>63</y>
         <width>70</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Fullscreen</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="buildTerrainPushButton">
      <property name="geometry">
       <rect>
        <x>8</x>
        <y>550</y>
        <width>159</width>
        <height>40</height>
       </rect>
      </property>
      <property name="text">
       <string>Build Terrain</string>
      </property>
     </widget>
     <widget class="QFrame" name="frame_5">
      <property name="geometry">
       <rect>
        <x>7</x>
        <y>176</y>
        <width>159</width>
        <height>63</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QPushButton" name="pushButton_4">
       <property name="geometry">
        <rect>
         <x>31</x>
         <y>9</y>
         <width>44</width>
         <height>44</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/QtTerrainGenerator/Resources/Translate.png</normaloff>:/QtTerrainGenerator/Resources/Translate.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>30</width>
         <height>30</height>
        </size>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_5">
       <property name="geometry">
        <rect>
         <x>81</x>
         <y>9</y>
         <width>43</width>
         <height>44</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>993</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave_Heightmap"/>
    <addaction name="actionLoad_Heightmap"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionSave_Heightmap">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Save Heightmap</string>
   </property>
  </action>
  <action name="actionLoad_Heightmap">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Load Heightmap</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>TerrainVisualizationWidget</class>
   <extends>QWidget</extends>
   <header>terrainvisualizationwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionSave_Heightmap</sender>
   <signal>triggered()</signal>
   <receiver>MainForm</receiver>
   <slot>_saveHeightmap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad_Heightmap</sender>
   <signal>triggered()</signal>
   <receiver>MainForm</receiver>
   <slot>_loadHeightMap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>_saveHeightmap()</slot>
  <slot>_loadHeightMap()</slot>
 </slots>
</ui>
